### PR TITLE
Implement swagger docs and fix path validation

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
ref: #52

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]

File: cmd/config.go [hunk 1]
```diff
@@ -564,15 +564,16 @@ func (c *Config) ValidatePathEnd(pe *relayer.PathEnd) error {
...
+	if err != nil {
+		return err
+	}
+
 	// if the identifiers are empty, don't do any validation
 	if pe.ClientID == "" && pe.ConnectionID == "" && pe.ChannelID == "" {
 		return nil
 	}
 
-	chain, err := c.Chains.Get(pe.ChainID)
```
Comment (colin-axner) [inline cmd/config.go:572]: is this change intentional? The previous check is cheaper, and prevents a panic if the user hasn't added the chain but also hasn't set path end information (and therefore doesn't need validation)

Comment (akhilkumarpilli) [reply cmd/config.go:572]: Yeah, this is intentional. When given client, connection and channel identifiers are empty, it will return nil and path will be created. But we are not checking chains are configured or not. This also throwing `chain not configured` error when querying paths.

That's the reason, I moved this check before checking all identifiers are empty.

Comment (colin-axner) [reply cmd/config.go:572]: I'm confused. Do any configuration checks occur if the client, connection, and channel id are all empty? It is my understanding that if all the identifiers are empty, there is no configuration validation that needs to be done. Setting all empty identifiers is a valid starting place on the relayer

Comment (akhilkumarpilli) [reply cmd/config.go:572]: Yeah, Okay. But can chain-id be empty as we need to set chain before setting path?

Comment (colin-axner) [reply cmd/config.go:572]: Ahh, chain ID should be set. Can you just add a check that pe.ChainID != "" as the first if statement in this function? This way we return an error instead of panicing

**Review by colin-axner — APPROVED** [review]
great work! Thanks
